src/main/java/com/cloudbees/hudson/plugins/folder/ItemDeletion.java

```diff
@@ -1,208 +0,0 @@
-/*
- * The MIT License
- *
- * Copyright (c) 2017 CloudBees, Inc.
- *
- * Permission is hereby granted, free of charge, to any person obtaining a copy
- * of this software and associated documentation files (the "Software"), to deal
- * in the Software without restriction, including without limitation the rights
- * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
- * copies of the Software, and to permit persons to whom the Software is
- * furnished to do so, subject to the following conditions:
- *
- * The above copyright notice and this permission notice shall be included in
- * all copies or substantial portions of the Software.
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
- * THE SOFTWARE.
- */
-package com.cloudbees.hudson.plugins.folder;
-
-import hudson.Extension;
-import hudson.ExtensionList;
-import hudson.model.Action;
-import hudson.model.Item;
-import hudson.model.Queue;
-import hudson.model.queue.SubTask;
-import hudson.model.queue.Tasks;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
-import java.util.concurrent.locks.ReadWriteLock;
-import java.util.concurrent.locks.ReentrantReadWriteLock;
-import javax.annotation.CheckForNull;
-import javax.annotation.Nonnull;
-import javax.annotation.concurrent.GuardedBy;
-import org.kohsuke.accmod.Restricted;
-import org.kohsuke.accmod.restrictions.NoExternalUse;
-
-/**
- * A {@link Queue.QueueDecisionHandler} that blocks items being deleted from entering the queue.
- *
- * @since TODO
- */
-@Restricted(NoExternalUse.class) // TODO remove once baseline core has JENKINS-35160
```

Comment:
Assuming that `Restricted` didn't work in plugin POMs for a while, maybe we need to reconfirm there are bogus usages in dependencies

Reply:
This was being used as an extension only. The new core baseline contains the identical class.